core/secrets.py

```
@@ -0,0 +1,574 @@
+    mask_in_logs: bool = True
+
+
+class SecretsManager:
+    """
+    Centralized secrets management system.
+
+    Provides secure access to secrets with validation, type conversion,
+    and protection against accidental logging.
+    """
+
+    def __init__(self, environment: str = None):
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)